### ISSUE TITLE
Two images, which loss weights is better.

### ISSUE THREAD

**Issue by AchoWu** [FlyMyAI/flymyai-lora-trainer#28]
Thank you for your wonderful project!
I found some problems with the loss weights [here.](https://github.com/FlyMyAI/flymyai-lora-trainer/blob/c092784149efa8985faa80ed6719e37d3c6f73b5/train.py#L264)
```
               weighting = compute_loss_weighting_for_sd3(weighting_scheme="none", sigmas=sigmas)
```
weighting_scheme="none". 
while diffusers.training_utils.compute_loss_weighting_for_sd3
```
def compute_loss_weighting_for_sd3(weighting_scheme: str, sigmas=None):
    """
    Computes loss weighting scheme for SD3 training.

    Courtesy: This was contributed by Rafie Walker in https://github.com/huggingface/diffusers/pull/8528.
    
    SD3 paper reference: https://huggingface.co/papers/2403.03206v1.
    """
    if weighting_scheme == "sigma_sqrt":
        weighting = (sigmas**-2.0).float()
    elif weighting_scheme == "cosmap":
        bot = 1 - 2 * sigmas + 2 * sigmas**2
        weighting = 2 / (math.pi * bot)
    else:
        weighting = torch.ones_like(sigmas)
    return weighting
```
So weighting is ones_like tensor.
I tried to test MMDIT's CFM weighting as follows:
```
def compute_loss_weighting(sigmas, scheme="rf", eps=1e-6):
    """
    sigmas: tensor of shape [B, 1, 1, 1, 1], noise scales
    scheme: str, one of ["rf"]
    """
    if scheme == "rf":
        # Rectified Flow: w_t = (b_t^2) / (-2 * lambda'_t)
        # a_t = 1 - sigmas, b_t = sigmas,
        t = sigmas  # for RF, sigma = t
        # lambda'_t = 2*(-1/(1-sigmas)-(1/sigmas))
        lambda_prime = 2 * (-1.0 / (1 - t + eps) - 1.0 / (t + eps))
        w = (-t / (2.0 * lambda_prime + eps)) ** 2
        return w
```

train data
<img width="879" height="541" alt="Image" src="https://github.com/user-attachments/assets/104d7f34-6069-4f29-953c-e55ba9b6ca8e" />
prompt
```
1girl, black_hair, brown_eyes, earrings, freckles, grey_background, jewelry, lips, long_hair, looking_at_viewer, nose, piercing, realistic, red_lips, solo, upper_body
```
Qwen-Image origin output

<img width="1104" height="1472" alt="Image" src="https://github.com/user-attachments/assets/7b7248d9-a4af-41d2-b56e-4c65d33dce77" />

compute_loss_weighting_for_sd3 lora output

<img width="1024" height="1024" alt="Image" src="https://github.com/user-attachments/assets/7556bcae-3239-4bc9-aaf1-925a473fa3dd" />

CFM weighting  lora output
<img width="1024" height="1024" alt="Image" src="https://github.com/user-attachments/assets/f496e699-8cf0-4830-9b3b-70ad743a79b8" />

CFM weighting retains the original output features of the model while integrating the training data features.

**Comment by akk-123** [comment]
@AchoWu Where is the reference for the CFM weighting code implementation?

**Comment by AchoWu** [comment]
> [@AchoWu](https://github.com/AchoWu) Where is the reference for the CFM weighting code implementation?

ABOVE 
```
def compute_loss_weighting(sigmas, scheme="rf", eps=1e-6):
    """
    sigmas: tensor of shape [B, 1, 1, 1, 1], noise scales
    scheme: str, one of ["rf"]
    """
    if scheme == "rf":
        # Rectified Flow: w_t = (b_t^2) / (-2 * lambda'_t)
        # a_t = 1 - sigmas, b_t = sigmas,
        t = sigmas  # for RF, sigma = t
        # lambda'_t = 2*(-1/(1-sigmas)-(1/sigmas))
        lambda_prime = 2 * (-1.0 / (1 - t + eps) - 1.0 / (t + eps))
        w = (-t / (2.0 * lambda_prime + eps)) ** 2
        return w
```
This is my implementation.

**Comment by ValentinKovalev** [comment]
Thank you for your contribution. This will be added in an upcoming release.
